doc/app/user_function.rst

```diff
@@ -1,102 +1,167 @@
-Turn your functions into composable apps
-========================================
+Writing your own apps
+=====================
 
-Just use the ``define_app`` decorator! This generates a wrapper class that has a reference to your function and can then become part of a composed function.
+.. When writing an app, consider what type of processing it will do, whether it should be "composable", and whether it makes sense to write it as a function or a class.
 
```

Comment:
It's worth defining the term composable from the start "A composable app is a process that can be added to a chain of other apps as part of a sequence of processes to apply in order.

example: `my_steps = step_1 + step_2 +step_3` "

Reply:
that's great!